agent/src/main/java/com/teamscale/jacoco/agent/AgentBase.java

```diff
@@ -87,9 +88,11 @@ public static void premain(String options, Instrumentation instrumentation) thro
 		HttpUtils.setShouldValidateSsl(agentOptions.shouldValidateSsl());
 
 		logger.info("Starting JaCoCo's agent");
-		org.jacoco.agent.rt.internal_43f5073.PreMain.premain(agentOptions.createJacocoAgentOptions(), instrumentation);
+		JacocoAgentBuilder agentBuilder = new JacocoAgentBuilder(agentOptions);
```

Comment:
For reviewer: created the AgentBuilder class to resolve AgentOptions growing over 300SLOC